### PR TITLE
Store BaggageContext in NoOpSpan

### LINKED ISSUE #138: NoOp Tracing/Span must propagate context
Otherwise we drop baggage which is used to propagate "just logging".

See also https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#behavior-of-the-api-in-the-absence-of-an-installed-sdk

### PR DESCRIPTION
**Pull request by slashmo**
Closes #138

### CODE REVIEW

**Review by ktoso — APPROVED** [review]
Could get a test?

That without enabling any tracer we still propagate context.

**Comment by slashmo** [comment]
> Could get a test?

Good point, will do 👌

File: Tests/TracingTests/InstrumentationSystemTests.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import Tracing
+import XCTest
+
+final class InstrumentationSystemTests: XCTestCase {
+    func testItProvidesAccessToATracer() {
+        let tracer = TestTracer()
+
+        XCTAssertNil(InstrumentationSystem.tracer(of: TestTracer.self))
+
+        InstrumentationSystem.bootstrapInternal(tracer)
```
Comment (ktoso) [inline Tests/TracingTests/InstrumentationSystemTests.swift]: good, was just about to comment on this, previous test version used `bootstrap()` which would have been wrong;

do we also need a way to "unset" it after the test runs or not hmmmm?

Comment (slashmo) [reply Tests/TracingTests/InstrumentationSystemTests.swift]: Yeah, should we just make the `bootstrapInternal` method accept an optional? Or is there an opposite better describing it?

Comment (ktoso) [reply Tests/TracingTests/InstrumentationSystemTests.swift]: accepting optional sounds good to me, good enough for our tests and similar use cases for a "reset" 👍

Comment (slashmo) [reply Tests/TracingTests/InstrumentationSystemTests.swift]: Done ✅

**Comment by ktoso** [comment]
Nice thanks!

**Pull request merged**